Automattic/newspack-newsletters
# feat: enable custom footer

### All Submissions:






### Changes proposed in this Pull Request:

Newsletter authors will sometimes want to create a custom footer instead of the default one Mailchimp inserts. As described in https://mailchimp.com/help/customize-your-footer-content/, Mailchimp will insert the custom footer unless a series of legally required merge fields are present. In the current state of the plugin this doesn't work because links containing merge fields are mangled, through prepending scheme and appending `utm_medium` parameter. For example:

```
<a href="*|ABOUT_LIST|*">Why did I get this email?</a>
```

will be converted to:

```
<a href="http://*|ABOUT_LIST|*?utm_medium=email">Why did I get this email?</a>
```

Due to the invalidity of the link, Mailchimp inserts the default footer.

This branch resolves this by filtering the processed link. The Mailchimp class returns clean links when the href contains a merge field.

Closes https://github.com/Automattic/newspack-newsletters/issues/366.

### How to test the changes in this Pull Request:

1. On `master`, create a Newsletter. At the end, add paragraphs containing each of the following (use the Gutenberg editor to set the links):
```
This email was sent to <a href="mailto:*|EMAIL|*">*|EMAIL|*</a>.
<a href="http://*|ABOUT_LIST|*">Why did I get this email?</a>
<a href="http://*|UNSUB|*">Unsubscribe from this list</a>
<a href="*|UPDATE_PROFILE|*">Update subscription preferences</a></p>
<p>*|LIST:ADDRESSLINE|*</p>
```

2. Send a test email, observe the markup is present and some of the merge fields are not replaced (this is normal).
3. Send the campaign. In emails, observe the markup is present but Mailchimp inserts the default footer anyway. Inspect the email markup and observe unwanted `http://` and `utm_source` in the footer links.
4. Switch to this branch and repeat the above.
5. The test email should be basically the same.
6. The sent campaign emails should not contain the default Mailchimp footer.
7. Verify all links in the custom footer work as expected.

### Other information:
